qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -113,26 +118,50 @@ flush()
     ceph --admin-daemon ${TEMPDIR}/rbd-mirror.asok ${cmd}
 }
 
-wait_for_image_replay_started()
+test_image_replay_state()
 {
-    local image=$1
-    local image_id s
+    local image_id=$1
+    local test_state=$2
+    local current_state=stopped
 
     test -n "${RBD_MIRROR_ASOK}"
 
+    ceph --admin-daemon ${RBD_MIRROR_ASOK} help | fgrep "${image_id}" &&
+	current_state=started
+    test "${test_state}" = "${current_state}"
+}
+
+wait_for_image_replay_state()
+{
+    local image=$1
+    local state=$2
+    local image_id s
+
     image_id=$(remote_image_id ${image})
     test -n "${image_id}"
 
     # TODO: add a way to force rbd-mirror to update replayers
-
     for s in 1 2 4 8 8 8 8 8 8 8 8; do
 	sleep ${s}
-	ceph --admin-daemon ${RBD_MIRROR_ASOK} help | grep "${image_id}" &&
-	    return 0
+	test_image_replay_state "${image_id}" "${state}" && return 0
     done
     return 1
 }
 
+wait_for_image_replay_started()
+{
+    local image=$1
+
+    wait_for_image_replay_state ${image} started
+}
+
+wait_for_image_replay_stopped()
```

Comment:
Yes, I was going to use this to test live image replayer removal after an image is removed (journaling is disabled). But right now I can't remove the image when it is replayed due to "Device or resource busy" error. Still I think it is desirable feature and we will be able to do this in some future, and this function will be useful then.